common/dynamicconfig/configstore/config_store_client.go

```diff
@@ -235,7 +237,11 @@ func (csc *configStoreClient) GetDurationValue(
 }
 
 func (csc *configStoreClient) UpdateValue(name dc.Key, value interface{}) error {
-	return csc.updateValue(name, value, csc.config.UpdateRetryAttempts)
+	dcValues, ok := value.([]*types.DynamicConfigValue)
```

Comment:
tbh this feels kinda odd, but I suppose you've run it and it's correct?

explicitly declaring the type is at least MUCH clearer about what's happening, so 👍👍  on that anyway.  I just wonder if converting to the correct type should be a responsibility of the caller of UpdateValue....